images/cilium/Dockerfile

```diff
@@ -47,12 +47,24 @@ ARG LIBNETWORK_PLUGIN
 # Please do not add any dependency updates before the 'make install' here,
 # as that will mess with caching for incremental builds!
 #
-# TARGETARCH is an automatic platform ARG enabled by Docker BuildKit.
-#
-ARG TARGETARCH
-RUN make GOARCH=$TARGETARCH RACE=$RACE NOSTRIP=$NOSTRIP LOCKDEBUG=$LOCKDEBUG PKG_BUILD=1 V=$V LIBNETWORK_PLUGIN=$LIBNETWORK_PLUGIN \
-    SKIP_DOCS=true DESTDIR=/tmp/install build-container install-container \
-    licenses-all
+WORKDIR /go/src/github.com/cilium/cilium
+RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium --mount=target=/root/.cache,type=cache --mount=target=/go/pkg/mod,type=cache \
+    make GOARCH=${TARGETARCH} RACE=${RACE} NOSTRIP=${NOSTRIP} LOCKDEBUG=${LOCKDEBUG} PKG_BUILD=1 V=${V} LIBNETWORK_PLUGIN=${LIBNETWORK_PLUGIN} \
+    SKIP_DOCS=true DESTDIR=/tmp/install/${TARGETOS}/${TARGETARCH} build-container install-container-binary
+
+RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium --mount=target=/root/.cache,type=cache --mount=target=/go/pkg/mod,type=cache \
+    # install-bash-completion will execute the bash_completion script. It is
+    # fine to run this with same architecture as BUILDARCH since the output of
+    # bash_completion is the same for both architectures.
+    make GOARCH=${BUILDARCH} RACE=${RACE} NOSTRIP=${NOSTRIP} LOCKDEBUG=${LOCKDEBUG} PKG_BUILD=1 V=${V} LIBNETWORK_PLUGIN=${LIBNETWORK_PLUGIN} \
+    SKIP_DOCS=true DESTDIR=/tmp/install/${TARGETOS}/${TARGETARCH} install-bash-completion licenses-all && \
+    mv LICENSE.all /tmp/install/${TARGETOS}/${TARGETARCH}/LICENSE.all
+
+RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium --mount=target=/root/.cache,type=cache --mount=target=/go/pkg/mod,type=cache \
+    cp /go/src/github.com/cilium/cilium/images/cilium/init-container.sh \
+       /go/src/github.com/cilium/cilium/plugins/cilium-cni/cni-install.sh \
+       /go/src/github.com/cilium/cilium/plugins/cilium-cni/cni-uninstall.sh \
+       /tmp/install/${TARGETOS}/${TARGETARCH}
```

Comment:
Does this duplicate install scripting from the Makefiles? I'd prefer to have only one source of truth as much as possible and have the make target also work when cross-compiling. Would this be possible? Could the makefile also handle the bash completion install?

Reply:
We don't perform this step with `make install` as this is specific for a container image.